podcasts/IAPHelper.swift

```diff
@@ -417,3 +422,28 @@ private extension IAPHelper {
         Analytics.track(event, properties: properties)
     }
 }
+
+// MARK: - ServerHandler
+
+extension IAPHelper {
+    /// Acts as a proxy to the `ServerSettings` static methods the IAPHelper uses, and the `ApiServerHandler` calls
+    /// This allows for these to be injected and test, but keep the functionality the same.
+    class ServerHandler {
```

Comment:
This class approach is totally valid, but couldn't we create a protocol that has these methods, and the have an implemention forthe real calls and another for the mock. That way we will avoid subsclassing this on the tests/mocks

Reply:
Makes sense. Using a protocol also prevents the possibility of the tests using production data. 

Since I wasn't happy with the naming, I decided to split the dependencies into 2 protocols: Settings and Networking

Done here: 
- 7be4ab20b245339e494a0b1e2123b14f933bea9e
- 056b8c017bc6f509a78065834dac1c85ea759248